tests/test_gradient.py

```diff
@@ -9,31 +9,20 @@
 
 class TestGradient(object):
 
-    @pytest.mark.parametrize('space_order', [4])
-    @pytest.mark.parametrize('kernel', ['OT2'])
-    @pytest.mark.parametrize('shape', [(70, 80)])
-    def test_gradient_checkpointing(self, shape, kernel, space_order):
+    @pytest.mark.parametrize('dtype, opt', [
+        (np.float32, ('noop', {'openmp': True, 'par-dynamic-work': 1000})),
+        (np.float64, 'advanced')
+    ])
+    def test_gradient_checkpointing(self, dtype, opt):
         r"""
-        This test ensures that the FWI gradient computed with devito
-        satisfies the Taylor expansion property:
-        .. math::
-            \Phi(m0 + h dm) = \Phi(m0) + \O(h) \\
-            \Phi(m0 + h dm) = \Phi(m0) + h \nabla \Phi(m0) + \O(h^2) \\
-            \Phi(m0) = .5* || F(m0 + h dm) - D ||_2^2
-
-        where
-        .. math::
-            \nabla \Phi(m0) = <J^T \delta d, dm> \\
-            \delta d = F(m0+ h dm) - D \\
-
-        with F the Forward modelling operator.
+        This test ensures that the FWI gradient computed with checkpointing matches
+        the one without checkpointing. Note that this test fails with dynamic openmp
+        scheduling enabled so this test disables it.
         """
-        spacing = tuple(10. for _ in shape)
-        wave = setup(shape=shape, spacing=spacing, dtype=np.float64,
-                     kernel=kernel, space_order=space_order,
-                     nbl=40)
+        wave = setup(shape=(70, 80), spacing=(10., 10.), dtype=dtype,
+                     kernel='OT2', space_order=4, nbl=40, opt=opt)
 
-        v0 = Function(name='v0', grid=wave.model.grid, space_order=space_order)
+        v0 = Function(name='v0', grid=wave.model.grid, space_order=4)
```

Comment:
I'd keep parametrization over `space_order` for future extensions

Reply:
I'm also unresolving this one. I think we should reinstate `space_order` for parametrization and also add at least one test case for higher space order